projects/vdk-plugins/vdk-duckdb/src/vdk/plugin/duckdb/ingest_to_duckdb.py

```diff
@@ -0,0 +1,207 @@
+                errors.log_and_rethrow(
+                    errors.ResolvableBy.PLATFORM_ERROR,
+                    log,
+                    "Failed to sent payload",
+                    "Unknown error. Error message was : " + str(e),
+                    "Will not be able to send the payload for ingestion",
+                    "See error message for help ",
+                    e,
+                    wrap_in_vdk_error=True,
+                )
+
+    def __check_destination_table_exists(
+        self, destination_table: str, cur: duckdb.cursor
+    ) -> None:
+        if not self._check_if_table_exists(destination_table, cur):
+            errors.log_and_throw(
+                errors.ResolvableBy.USER_ERROR,
+                log,
+                "Cannot send payload for ingestion to DuckDB database.",
+                "destination_table does not exist in the target database.",
+                "Will not be able to send the payloads and will throw exception."
+                "Likely the job would fail",
+                "Make sure the destination_table exists in the target DuckDB database.",
+            )
+
+    @staticmethod
+    def _check_if_table_exists(table_name: str, cur: duckdb.cursor) -> bool:
+        cur.execute("show tables")
+        tables = cur.fetchall()
+        return (table_name,) in tables
+
+    def __table_columns(
+        self, cur: duckdb.cursor, destination_table: str
+    ) -> List[Tuple[str, str]]:
+        columns = []
+        if self._check_if_table_exists(destination_table, cur):
+            for row in cur.execute(
+                f"SELECT column_name, data_type FROM information_schema.columns WHERE table_name = '{destination_table}'"
+            ).fetchall():
+                columns.append((row[0], row[1]))
+        return columns
+
+    def __create_query(
+        self, destination_table: str, payload: List[dict], cur: duckdb.cursor
+    ) -> Tuple[list, str]:
+        fields = [
+            field_tuple[0]
+            for field_tuple in cur.execute(
+                f"SELECT column_name FROM information_schema.columns WHERE table_name = '{destination_table}'"
+            ).fetchall()
+        ]
+
+        for obj in payload:
+            if collections.Counter(fields) != collections.Counter(obj.keys()):
+                errors.log_and_throw(
+                    errors.ResolvableBy.USER_ERROR,
+                    log,
+                    "Failed to sent payload",
+                    f"""
+                    One or more column names in the input data did NOT
+                    match corresponding column names in the database.
+                       Input Table Columns: {list(obj.keys())}
+                    Database Table Columns: {fields}
+                    """,
+                    "Will not be able to send the payload for ingestion",
+                    "See error message for help ",
+                )
+
+        values = [[obj.get(field) for field in fields] for obj in payload]
```

Comment:
Not sure the next three lines are entirely necessary. I added them to the sqlite plugin, because it was being a bit stupid when it came to whitespace in column names. https://github.com/vmware/versatile-data-kit/pull/2510

You can check how duckdb handles whitespace in column names and maybe simplify the behaviour.